### ISSUE TITLE
[UX] On the Add layout form, move "path" and "visibility conditions" above the template icons.

### ISSUE THREAD

**Issue by docwilmot** [backdrop/backdrop-issues#1606]
Because when you have a bunch of layout templates (or layouts or whatever they are; #998, sigh..) form errors aren't visible.

If you try to add a `Path` or a `visibility condition` before you add a layout name, the form errors pop up via AJAX at the top of the page. But if you have a bunch of layout (templates), because you're at the bottom of the page you wouldn't know that and might think things are behaving weird.

So put the grid of layout (templates) last so all the form field input stuff is near the error mesages.

**Comment by klonos** [comment]
...was about to file an issue for that myself just now. As you can see in the screencast below, the form validation messages when you hit the "Add visibility condition" link are outside the view port. This leads the user to assume that there is something wrong with the link:

![backdrop-issue1606-layout_add_form_validation_errors_outside_viewport](https://cloud.githubusercontent.com/assets/2423362/12869750/3cf12d3c-cd7c-11e5-8827-e0816c9d16e0.gif)

**Comment by klonos** [comment]
...we could alternatively disable/hide the "Add visibility condition" link while there are missing required fields and only show it when these fields are filed in. Or replace it with a "You need to specify a layout name, path and choose a *layout template* before you can add visibility conditions." text (*layout template* part pending decision over at #998).

BTW, I have spotted a few more annoyances in this form while testing things out. About to file a series of new issues for them instead of listing them here and risking getting OT...

**Comment by klonos** [comment]
...the reason I propose alternatives to moving things around is not to dis your suggestion @docwilmot. I realize that in most other cases this would be the right solution (to move the input elements closer to the error message that is).

In this case though, we have placed input elements in the order that people do things in order to create a new layout ...a step-by-step wizard thing as they scroll down the page if you like. Path selection is secondary in the process compared to selecting how the layout looks and visibility conditions are an optional thing. Moving them above a required element (layout template selection) is wrong and gives it unnecessary importance.

**Comment by docwilmot** [comment]
Agreed @klonos. I think maybe moving the `Path` field is probably OK, but leave the `Add condition` where it is. And either disable the link till required fields are filled and/or scroll the page to top when errors pop up and/or put the errors down there by the link. (consider [Inline Form Errors](https://www.drupal.org/project/ife)

**Comment by klonos** [comment]
Yep, that sounds about right. An alternative (of the alternative) would be to use either vertical or horizontal tabs. One for the layout name, path and visibility (call it layout settings) and a separate for the layout designs/templates (though I am not particularly fond of that idea - just putting it out there). This would ensure that any error messages would not be way up at the top of the page where people need to scroll in order to see them (since the bulk of layout templates/designs would be moved to the other tab).

**Comment by klonos** [comment]
...now that I've mentioned it, the issue here IS with the list of templates/designs and the way they push the visibility conditions section and the form submit buttons to the bottom, away from the error message area. I know that we've greatly improved the UX by converting the list of templates/designs to something that users can see in some sort of preview, but at the same time we've created this usability issue here.

So, how about this alternative (yet another)? ...what if we moved the template/design list to a modal and replaced the whole section with a "No layout design selected" text and a "Select design" button/link? Once a layout design is selected, then we can show only that one thumbnail (plus its name of course) and and change the button/link to say "Change layout design".

Thoughts?

**Comment by klonos** [comment]
> ...what if we moved the template/design list to a modal...

In the layout edit page, there's the "Settings" tab that holds only the list of templates/designs. If we were to move them to a dialog, then there would be no need for this tab any more.

**Comment by BWPanda** [comment]
Sounds like this form could use some improvements as discussed already, but I'd like to get feedback on whether an alternative fix for the original issue here would be to simply display error messages above the field they refer to...?

For AJAX stuff this would prevent the need for scrolling at all, and for page refreshes we could display a message at the top saying there are validation errors and then provide the specific errors directly above the fields they relate to.

If others think this is worth pursuing, then I can create a separate, generic issue for it.

**Comment by klonos** [comment]
>If others think this is worth pursuing, then I can create a separate, generic issue for it.

It is definitely worth pursuing @BWPanda! ...that's why we already have a separate issue for it: #1040 😉

**Comment by BWPanda** [comment]
Ha, nice!
